### DISCUSSION TITLE
how to add spaces between the address and the instruction

### DISCUSSION THREAD

**Discussion by 440bx** [zyantific/zydis#493] [Q&A] [answered]
Hello, 

I'd like to put more spaces between the address and the instruction.  I'm guessing that I probably need to hook one of the ZYDIS_FORMATTER_FUNC.... functions but, I am not sure which one.

The questions are: is a formatter function hook the only/simplest way of controlling the number of spaces between the address and the instruction text ?... if it isn't what's the simplest way, if it is then which function should I hook ?

To make it crystal clear what I'm trying to do, I want to go from this (the default output):
"  7FFFFFFF400007   push rax" to this:
"  7FFFFFFF400007...............................push rax"  (note: the dots should be spaces not dots)

Thank you for your help.

**Comment by flobernd** [comment, marked as answer]
The leading address is not really part of the formatter output. It's just manually written:
https://github.com/zyantific/zydis/blob/ffde0f46398a86417c860462f6af0556331cb5bb/examples/DisassembleSimple.c#L60

You can add as many spaces to the format specifier string template as you like.

**Reply by 440bx** [reply]
Couldn't be any simpler!! perfect, thank you very much and apologies for asking a dumb question, I should have seen that. !!
